redis/src/types.rs

```diff
@@ -1398,6 +1457,14 @@ impl FromRedisValue for String {
             _ => invalid_type_error!(v, "Response type not string compatible."),
         }
     }
+    fn from_owned_redis_value(v: Value) -> RedisResult<String> {
```

Comment:
As far as I understood, `CString` can't necessarily be implemented efficiently, because it's necessary to append the null byte to the allocation during the conversion. If the string/byte arrays had their size reserved exactly when the `redis::Value` was built (I haven't checked this but it seemed plausible) then copying into the new `CString` will be basically the same performance-wise as attempting to mutate the contents to add the null and then do some kind of fancy cast.

Reply:
Okay, I looked into this a bit more - basically `CString` will do its best, so can just defer to passing a `String` or `Vec<u8>` in here too. Will implement this case.